src/cargo/ops/cargo_remove.rs

```diff
@@ -46,20 +58,118 @@ pub fn remove(options: &RemoveOptions<'_>) -> CargoResult<()> {
 
         manifest.remove_from_table(&dep_table, dep)?;
 
-        // Now that we have removed the crate, if that was the last reference to that
-        // crate, then we need to drop any explicitly activated features on
-        // that crate.
+        // Now that we have removed the crate, if that was the last reference to that crate, then we
+        // need to drop any explicitly activated features on that crate.
         manifest.gc_dep(dep);
     }
 
+    // remove from workspace dependencies
+
+    let mut ws_manifest = None;
+
+    if !to_remove_workspace.is_empty() {
+        // read the packages in the workspace, substituting in the modified manifest
+        let members = options
+            .workspace
+            .members()
+            .map(|p| {
+                if p.manifest_path() == manifest_path {
+                    Ok((p, manifest.clone()))
+                } else {
+                    LocalManifest::try_new(p.manifest_path()).map(|m| (p, m))
+                }
+            })
+            .collect::<CargoResult<Vec<_>>>()?;
```

Comment:
For some reason I'm not seeing where the lock-file update is happening but we could just update the lock file before this and read that.  `workspace.dependencies` has no affect on the lockfile; we dropped that part of the RFC

Reply:
Lock file updates happen in `src/bin/cargo/commands/remove.rs`, like cargo add.

Reply:
Thinking about this some more, I lean towards doing workspace GC at the same level as locking the workspace.

As a side note, we'll probably need to continue to parse all of the manifest members to see if the dependency is unused since a manifest might declare a dependency that is similar to a workspace dependency without using it.

Reply:
I just pushed a version with a separate `gc_workspace` function for reference. Still using the "predicting" approach, so if we wanted to push it up a level, we'd want to also push up that `to_remove` list.

Reply:
Okay, [compare e99de7c and 8022dda](https://github.com/rust-lang/cargo/compare/e99de7c74a71264ef4cbd0fd3045ef77157a149c..8022ddaaa6a2a7ef91ac7c77c6373ba8c79eaf47) to see the functionality pushed up. The main difference is that now the GC operation isn't aware of which dependencies were marked `workspace = true`, so it just assumes that all of them were and reads every manifest just in case.

Reply:
Thanks!  I feel like that is a big help.  We can always move `gc_workspace` into an `op` if needed at a later point.

>  The main difference is that now the GC operation isn't aware of which dependencies were marked workspace = true, so it just assumes that all of them were and reads every manifest just in case.

I think thats fine.  The overhead of checking is probably low enough in a command like this.